app/uk/gov/hmrc/transitmovements/repositories/MovementsRepository.scala

```diff
@@ -525,8 +534,8 @@ class MovementsRepositoryImpl @Inject() (
         Future.successful(Left(UnexpectedError(Some(ex))))
     })
 
-  def restrictDuplicateLRN(localReferenceNumber: LocalReferenceNumber): EitherT[Future, MongoError, Unit] = {
-    val selector = mEq("localReferenceNumber", localReferenceNumber.value)
+  def restrictDuplicateLRN(localReferenceNumber: LocalReferenceNumber, messageSender: MessageSender): EitherT[Future, MongoError, Unit] = {
+    val selector = mAnd(mEq("localReferenceNumber", localReferenceNumber.value), mEq("messageSender", messageSender.value))
```

Comment:
This query will need to be expanded on slightly -- from the ticket:

> ... OR if the LRN has been used in a departure with no message sender, so that we still catch all duplicates

So, this query needs to return if there are any entries if:

* the LRN and messageSender exist on another document (which is this document), OR
* the LRN exists on a document WITHOUT a messageSender (so `mAnd(mEq("localReferenceNumber"), mNot(mExists("messageSender")))`)

Note that `mNot` and `mExists` would need to be aliased from `Filters.not` and `Filters.exists`.

We need this as the database doesn't currently hold message senders, so we don't want to let potential duplicate LRNs through.

Reply:
Yes correct, updated it please check